freqtrade/optimize/hyperopt_loss_sharpe_daily.py

```diff
@@ -0,0 +1,58 @@
+"""
+SharpeHyperOptLoss
+
+This module defines the alternative HyperOptLoss class which can be used for
+Hyperoptimization.
+"""
+from datetime import datetime
+
+from pandas import DataFrame
+import numpy as np
+
+from freqtrade.optimize.hyperopt import IHyperOptLoss
+
+
+class SharpeHyperOptLossDaily(IHyperOptLoss):
+    """
+    Defines the loss function for hyperopt.
+
+    This implementation uses the Sharpe Ratio calculation.
+    """
+
+    @staticmethod
+    def hyperopt_loss_function(
+        results: DataFrame,
+        trade_count: int,
+        min_date: datetime,
+        max_date: datetime,
+        *args,
+        **kwargs
+    ) -> float:
+        """
+        Objective function, returns smaller number for more optimal results.
+
+        Uses Sharpe Ratio calculation.
+        """
+        total_profit = results.profit_percent
+
+        # adding slippage of 0.1% per trade
+        total_profit = total_profit - 0.0005
```

Comment:
This does not seem to be correct.
While it's aligned to `SharpeHyperOptLoss` - `total_profit` is not used after this point - which makes this line pointless.

There's 2 ways to correct this i think.
Either, using `total_profit` throughout instead of `results['profit_percent']`
or - modify the column of the results dictionary as follows.

``` python
results.loc[:, 'profit_percent'] = results['profit_percent'] - 0.0005
```

Reply:
So replacing this line `total_profit = total_profit - 0.0005` with `results.loc[:, 'profit_percent'] = results['profit_percent'] - 0.0005` would work?

Reply:
@xmatthias I'm now thinking it should be like this:
`results.loc[:, 'profit_percent'] = results.loc[:, 'profit_percent'] - 0.0005`

Reply:
`results.loc[:, 'profit_percent'] = results.loc[:, 'profit_percent'] - 0.0005` is technically identical to the above `total_profit = total_profit - 0.0005` line.
this selects the column profit_percent- and reassigns it to itself with 0.0005 subtracted. 

the `total_profit` version selects the column `profit_percent` in line 36 - and then does the same thing.

This works in the "old" hyperopt_loss_sharpe method, as `results` is not reused after that.
I like the resample on top of `results` better as it's clearer what the type of this is - so i think you could remove line 36, change this to `results.loc[:, 'profit_percent'] = results['profit_percent'] - 0.0005` - and it'll have the same effect.

Reply:
Got it, thanks for the explanation, I'm actually running a course right now on datacamp that teaches how to play with pandas dataframe.